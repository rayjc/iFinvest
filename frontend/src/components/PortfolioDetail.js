import React from 'react';
import { useParams } from 'react-router-dom';
import InvestmentsList from './InvestmentsList';
import NewInvestment from './NewInvestment';
import PortfolioChart from './PortfolioChart';


const PortfolioDetail = () => {
  const { id } = useParams();

  return (
    <div style={{ paddingBottom: '80px' }}>
      <InvestmentsList portfolioId={+id} />
      <NewInvestment portfolioId={+id} />
      <PortfolioChart portfolioId={+id} />
    </div>
  );

};


export default PortfolioDetail;