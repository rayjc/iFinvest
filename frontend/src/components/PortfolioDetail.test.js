import React from 'react';
import { useParams, MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import PortfolioDetail from './PortfolioDetail';
import { useSelector, useDispatch } from 'react-redux';


const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: 0
  }),
}));

beforeEach(function() {

  useSelector.mockImplementation((selectorFn) =>
    selectorFn({
      user: {
        user: {
          id: 0, username: "test", first_name: "test", last_name: "user",
          email: "test@test.com"
        },
        error: null, isFetching: false,
      },
      portfolios: {
        portfolios: [
          {
            id: 0, name: "alpha", user_id: 0, created_at: "2020-01-10",
            investments: [{ id: 0, symbol: "test" }]
          },
        ],
        error: null, isFetching: false,
      },
      investments: {
        investments: {}, error: null, isFetching: false,
      }
    }
    )
  );
});


// smoke test
it("renders without crashing", function() {
  render(<MemoryRouter><PortfolioDetail /></MemoryRouter>);
});

// snapshot test
it("matches snapshot", function() {
  const { asFragment } = render(<MemoryRouter><PortfolioDetail /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});