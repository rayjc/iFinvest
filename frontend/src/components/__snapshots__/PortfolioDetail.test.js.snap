// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style="padding-bottom: 80px;"
>
  <main
    class="MuiContainer-root makeStyles-root-13 MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-14"
    >
      <div
        class="MuiGrid-root makeStyles-title-15 MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            alpha
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            $0.00
          </h6>
        </div>
      </div>
      <div
        class="makeStyles-loading-20"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </main>
  <div
    style="text-align: center; margin: 20px;"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        New Investment
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="PortfolioChart"
  >
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root makeStyles-fab-23 MuiFab-extended MuiFab-secondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-extendedIcon-24"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
        Compute
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
